### PR TITLE
Add Activity API, docs and dredd tests

### LINKED ISSUE #4097: Add Activity API with relationships and docs
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-orga-server

### PR DESCRIPTION
**Pull request by enigmaeth**
Fixes #4097. 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in this pull request:

-
-
-

### CODE REVIEW

**Comment by niranjan94** [comment]
Activities should only allow read operations via the api. 

Activities should be internally created based on actions performed

**Comment by enigmaeth** [comment]
@niranjan94 Made changes so that GET can be accessed by anyone and admin access is required for POST. 
Also, help me with how we can create activities based on actions performed? Will this be limited to only those which are listed in the activities model, or for each request that is made?

**Comment by niranjan94** [comment]
You misunderstood. GET is accessible by admin. Post, patch, delete not
accessible by anyone.

**Comment by enigmaeth** [comment]
Okay. I see. Will update that.
I kept looking for ways to POST the activities today.

**Comment by enigmaeth** [comment]
@niranjan94 @shubham-padia Updated. Please review.

File: app/api/activities.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    action = fields.Str(allow_none=True)
+
+
+class ActivityList(ResourceList):
+    """
+    List and create activity
+    """
+    schema = ActivitySchema
+    methods = ['GET', ]
+    decorators = (api.has_permission('is_admin', methods="GET"),)
```
Comment (poush) [inline app/api/activities.py]: Not the issue but the **methods** is not required here since GET is the only accepted method in resource.

**Pull request merged**